### ISSUE TITLE
Decoder creates object property with undefined value

### ISSUE THREAD

**Issue by w01fgang** [nvie/decoders#573]
### The actual behavior
```javascript
const decoder = object({
    id: number,
    name: string,
    address: optional(string),
});
decoder({ id: 1, name: "test" }).unwrap();
// returns this
{ id: 1, name: "test", address: undefined };
```

### Expected behavior
```javascript
const decoder = object({
    id: number,
    name: string,
    address: optional(string),
});
decoder({ id: 1, name: "test" }).unwrap();
// returns this
{ id: 1, name: "test" };
```

### Why it matters?
```javascript
type Params = {|
  host?: string,
  port?: number,
|};
const decoder = exact({
  host: optional(string),
  port: optional(number),
});
const validate = guard(decoder);
const params = validate({});
const options = {
  host: "localhost",
  port: 3000,
  ...params,
};
console.log(options);
/*
{
  host: undefined,
  port: undefined
}
*/
```

### Workaround
```javascript
const decoder = exact({
  host: optional(string),
  port: optional(number),
});
const validate = guard(decoder);
const params = JSON.parse(JSON.stringify(validate({})));
console.log(params);
/*
{}
*/
```

**Comment by nvie** [comment]
Hi @w01fgang — thanks for reporting this! I've made a PR for this, and added some test cases to express the new behavior. I'll need to give this a pass over some real world code bases before I merge this in, because I want to make sure I don't break any behavior that is implicitly relied upon.

**Comment by w01fgang** [comment]
Hi @nvie, thanks a lot!
decoders are great, thank you for creating and sharing them!

**Comment by nvie** [comment]
Thanks @w01fgang – I just released this as [v1.23.3](https://github.com/nvie/decoders/releases/tag/v1.23.3) 🙏
